### PR TITLE
docs: update pytest supported scope

### PR DESCRIPTION
**Pull request by miketheman**
## Description

The reported pytest scope claims anything above 3.0, however the API
used of `Node.iter_markers(name)` was only introduced in pytest 3.6.0

Running this on earlier versions of pytest will error with (abbreviated
stack):

    INTERNALERROR>   File "/usr/local/lib/python3.6/site-packages/ddtrace/contrib/pytest/plugin.py", line 83, in pytest_runtest_protocol
    INTERNALERROR>     markers = [marker.kwargs for marker in item.iter_markers(name="dd_tags")]
    INTERNALERROR> AttributeError: 'Function' object has no attribute 'iter_markers

This might benefit from a test scope tightening.

Refs: https://docs.pytest.org/en/stable/changelog.html#pytest-3-6-0-2018-05-23

Signed-off-by: Mike Fiedler <miketheman@gmail.com>

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by jirikuncar — DISMISSED** [review]

File: docs/index.rst [hunk 1]
```diff
@@ -118,7 +118,7 @@ contacting support.
 +--------------------------------------------------+---------------+----------------+
 | :ref:`pyramid`                                   | >= 1.7        | No             |
 +--------------------------------------------------+---------------+----------------+
-| :ref:`pytest`                                    | >= 3.0        | Yes            |
+| :ref:`pytest`                                    | >= 3.6.0      | Yes            |
```
Comment (Kyle-Verhoog) [inline docs/index.rst:121]:
```suggestion
| :ref:`pytest`                                    | >= 4.0.0      | Yes            |
```

It looks like we actually only test with 4 and above

**Review by Kyle-Verhoog — COMMENTED** [review]
Thanks @miketheman! I think we can bump the version up to 4 to be safer since that's the minimum that is actually tested.

I'll make a note for us to expand the test suite to see what minor versions we support

File: docs/index.rst (hunk 1 above)
Comment (miketheman) [reply docs/index.rst:121]: @Kyle-Verhoog Thanks! I was going off this:
https://github.com/DataDog/dd-trace-py/blob/eef75c3117dc378e9d548a95ea2291819e934757/tox.ini#L317-L318

is that incorrect?

Comment (Kyle-Verhoog) [reply docs/index.rst:121]: Ah yeah those are just the environment definitions (tox files are 🤯 to read). These are the lines that specify the environments for the pytest integration tests: https://github.com/DataDog/dd-trace-py/blob/eef75c3117dc378e9d548a95ea2291819e934757/tox.ini#L102-L103

**Review by Kyle-Verhoog — APPROVED** [review]

**Comment by Kyle-Verhoog** [comment]
thanks @miketheman!! 🙂

**Review by jd — APPROVED** [review]

**Pull request merged**